src/VisualStudio/Core/Def/SampleToolWindow/SampleToolWindowFactory.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.ComponentModel.Composition;
+using System.Diagnostics.CodeAnalysis;
+using System.IO.Packaging;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Editor;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.VisualStudio.LanguageServices.Setup;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices
+{
+    [Export(typeof(IWpfTextViewCreationListener))]
+    [ContentType(ContentTypeNames.RoslynContentType)]
+    [TextViewRole(PredefinedTextViewRoles.Document)]
+    [Export(typeof(SampleToolWindowFactory))]
+    internal class SampleToolWindowFactory : IWpfTextViewCreationListener
+    {
+        private bool _initialized;
+
+        [MemberNotNullWhen(true, nameof(_initialized))]
+        private RoslynPackage? Package { get; set; }
+        [MemberNotNullWhen(true, nameof(_initialized))]
+        private IThreadingContext? ThreadingContext { get; set; }
+
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public SampleToolWindowFactory()
+        {
+        }
+
+        public void Initialize(RoslynPackage package)
+        {
+            Package = package;
+            ThreadingContext = package.ComponentModel.GetService<IThreadingContext>();
+            _initialized = true;
+        }
+
+        /// <summary>
+        /// Called whenever a new file is opened
+        /// </summary>
+        public void TextViewCreated(IWpfTextView textView)
+        {
+            _ = ShowSampleToolWindowAsync();
```

Comment:
❗ All asynchronous work must be tracked, either in the life of the caller or via a listener.

Reply:
yep, you'll need to do something like `ThreadingContext.JoinableTaskFactory.RunAsync(() => ShowSampleToolWindowAsync()).FileAndForget("Openning Sample Tool Window");`